packages/shared/lib/contexts/governance/actions/registerProposalsFromPrimaryNode.ts

```diff
@@ -0,0 +1,34 @@
+import { get } from 'svelte/store'
+
+import { selectedAccount } from '@core/account/stores'
+import { activeProfile } from '@core/profile/stores'
+
+import { getVotingEventIds, isProposalAlreadyAddedForSelectedAccount } from '../utils'
+
+import { registerParticipationEvent } from './registerParticipationEvent'
+
+export async function registerProposalsFromPrimaryNode(): Promise<void> {
+    const proposalIds = await getVotingEventIds()
+    if (!proposalIds || proposalIds.length === 0) {
+        return
+    }
+
+    const clientOptions = get(activeProfile)?.clientOptions
+    if (!clientOptions) {
+        throw new Error('Unable to retrieve profile client options')
+    }
+
+    const primaryNode = clientOptions?.primaryNode ?? clientOptions?.nodes[0]
```

Comment:
We should just pass in the nodes, and we can call this in a wrapper function for all the nodes the user has.